### PR TITLE
Closes #537: Chart with mock data generated by controller

### LINKED ISSUE #537: Chart with mock data generated by controller
Communication with controller should be based on ajax.

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js [hunk 1]
```diff
@@ -1,80 +1,217 @@
...
     }
-	
-	
-	
-	
-	<!--$('input[id="daterange"]').daterangepicker();-->
-	
+    
+    /** Show a point tooltip on the main chart */
+    $("#mainChart").bind("plothover", function (event, pos, item) {
```
Comment (pawelczak) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js]: It's better to use `on` than `bind`.

"As of jQuery 1.7, the .on() method is the preferred method for attaching event handlers to a document."
http://api.jquery.com/bind/

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js [hunk 2]
```diff
@@ -1,80 +1,217 @@
...
+     */
+    function printMainChart(chart, xmin, xmax) {
+
+        var yMinMax = calculateYAxisRangeForXRange(chart, xmin, xmax);
+        var ymin = yMinMax[0];
+        var ymax = yMinMax[1];
+        
+        var seriesArr = [];
+        
+        for (var i = 0; i < chart.seriesList.length; i++) {
```
Comment (pawelczak) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js]: Performance wise it is better to create variable for storing array length, than checking it every iteration. In this case you can write:
`for(var i = 0, length = chart.seriesList.length; i < length; i++) {`

Comment (lukdumi) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js]: OK, thanks for the advice and I will keep it in mind. But I googled a little about it and found that 
1. it is not a problem for new compilers 
2. it is not a problem for such a small array

And consider this color selection as something temporary, that will change in #565. That's why I think we can skip it now.

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js (hunk 1 above)
Comment (lukdumi) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js]: :+1:

**Pull request closed without merging**